### ISSUE TITLE
NullRef in SwiftUrlProvider.toUrl/sign

### ISSUE THREAD

**Issue by AliveDevil** [iterate-ch/cyberduck#13658]
Steps to reproduce:
- Connect to any bookmark, select a file
- Switch to bookmarks view, connect to Swift
- Observe crash

[637961585992787567.txt](https://github.com/iterate-ch/cyberduck/files/9336221/637961585992787567.txt)

**Comment by dkocher** [comment]
```
-----------------------------
[General Info]

Application: Cyberduck
Version:     8.4.3.38184
Region:      Deutsch (Deutschland)
Date: 2022-08-15
Time: 09:09

User Explanation:

User said ""
-----------------------------

[Exception Info 1]

Top-level Exception
Type:        System.NullReferenceException
Message:     Der Objektverweis wurde nicht auf eine Objektinstanz festgelegt.
Source:      Cyberduck.Protocols
Stack Trace: bei ch.cyberduck.core.openstack.SwiftUrlProvider.sign(Region region, Path file, Int64 expiry)
   bei ch.cyberduck.core.openstack.SwiftUrlProvider.toUrl(Path file)
   bei ch.cyberduck.core.vault.registry.VaultRegistryUrlProvider.toUrl(Path file)
   bei ch.cyberduck.core.DefaulPromptUrlProvider.isSupported(Path file, Type type)
   bei Ch.Cyberduck.Ui.Core.Commands.Command.Validate()
   bei Ch.Cyberduck.Ui.Core.Commands.Validate()
   bei System.EventHandler.Invoke(Object sender, EventArgs e)
   bei System.Windows.Forms.Application.ThreadContext.System.Windows.Forms.UnsafeNativeMethods.IMsoComponent.FDoIdle(Int32 grfidlef)
   bei System.Windows.Forms.Application.ComponentManager.System.Windows.Forms.UnsafeNativeMethods.IMsoComponentManager.FPushMessageLoop(IntPtr dwComponentID, Int32 reason, Int32 pvLoopData)
   bei System.Windows.Forms.Application.ThreadContext.RunMessageLoopInner(Int32 reason, ApplicationContext context)
   bei System.Windows.Forms.Application.ThreadContext.RunMessageLoop(Int32 reason, ApplicationContext context)
   bei Ch.Cyberduck.Ui.Program.Main(String[] args)
```

**Comment by dkocher** [comment]
Must check if reproducible with 71cff59d3ec74dc92109f048e27c599a63fc8efb.

**Comment by dkocher** [comment]
Can still be reproduced

```
Application: Cyberduck
Version:     8.5.3.38869
Region:      English (Switzerland)
Date: 15/12/2022
Time: 12:03

User Explanation:

User said ""
-----------------------------

[Exception Info 1]

Top-level Exception
Type:        System.NullReferenceException
Message:     Object reference not set to an instance of an object.
Source:      Cyberduck.Protocols
Stack Trace: at ch.cyberduck.core.openstack.SwiftLocationFeature.getLocation(Path file)
   at ch.cyberduck.core.openstack.SwiftRegionService.lookup(Path file)
   at ch.cyberduck.core.openstack.SwiftUrlProvider.toUrl(Path file)
   at ch.cyberduck.core.vault.registry.VaultRegistryUrlProvider.toUrl(Path file)
   at ch.cyberduck.core.DefaulPromptUrlProvider.isSupported(Path file, Type type)
   at Ch.Cyberduck.Ui.Core.Commands.Command.Validate()
   at Ch.Cyberduck.Ui.Core.Commands.Validate()
   at System.Windows.Forms.Application.ThreadContext.System.Windows.Forms.UnsafeNativeMethods.IMsoComponent.FDoIdle(Int32 grfidlef)
   at System.Windows.Forms.Application.ComponentManager.System.Windows.Forms.UnsafeNativeMethods.IMsoComponentManager.FPushMessageLoop(IntPtr dwComponentID, Int32 reason, Int32 pvLoopData)
   at System.Windows.Forms.Application.ThreadContext.RunMessageLoopInner(Int32 reason, ApplicationContext context)
   at System.Windows.Forms.Application.ThreadContext.RunMessageLoop(Int32 reason, ApplicationContext context)
   at Ch.Cyberduck.Ui.Program.Main(String[] args)
```

**Comment by dkocher** [comment]
Fix in 9a977f65.
